### ISSUE TITLE
"Group" Toggle

### ISSUE THREAD

**Issue by samvickars** [jacomyal/sigma.js#670]
Hello,

I'm relatively new to all of this, but I'm using the sigma.js exporter for Gephi and was wondering if there is a way to add somewhat of a toggle switch for groups of nodes?

Right now I have a group selector (like the one in this example: http://oxfordinternetinstitute.github.io/InteractiveVis/network/). Basically, I want to turn that group selector into a checkbox list of my groups, allowing the viewer to turn off or turn on each one.

Anybody know how to do this?

Cheers and thanks in advance.

**Comment by Yomguithereal** [comment]
Hello @samvickars. I do not know how you can alter the Oxford tool to fit your need specifically but what you intend to achieve is perfectly doable with sigma.

**Comment by samvickars** [comment]
Thanks for the reply @Yomguithereal. I figured it must be possible with sigma. Do you know how how? Could I possibly add this outside of the oxford tool, perhaps in my bootstrap header or similar?

**Comment by Yomguithereal** [comment]
Yes I know how but this would mean to build your own UI etc.

**Comment by samvickars** [comment]
Yes, I'm fine with doing that. Can you just show me the basics of how to build and incorporate it?

**Comment by Yomguithereal** [comment]
You can start by checking out some examples [here](https://github.com/jacomyal/sigma.js/tree/master/examples) (specifically the one called `basic`) and as soon as you can instantiate your graph I can show you how to filter nodes etc.

**Comment by samvickars** [comment]
I'm sure I can figure it out on my own. Thank you anyway.

**Comment by Yomguithereal** [comment]
Don't hesitate to pass by here if you need help with the code. I have snippets somewhere for camera handling, filtering etc. if you ever need them.

**Comment by SmokedEel** [comment]
@Yomguithereal I would be interested in looking at the snippets of code for filtering and camera handling.  Having a hard time understanding the filter

**Comment by Yomguithereal** [comment]
[Camera handling](https://github.com/Yomguithereal/agent-smith/blob/master/js/components/graph.jsx#L27-L53), zoom, unzoom, rescale.

**Comment by Yomguithereal** [comment]
[Here](https://github.com/medialab/manylines/blob/master/static/js/graph/graph.tasks.js) you have a series of functions dealing with graph operations.

**Comment by Yomguithereal** [comment]
To hide some nodes, for instance, you could do:

```js
s.graph.nodes([idsOfNodesINeedToHide]).forEach(function(node) {
  node.hidden = true;
});

s.refresh();
```
